sound/soc/intel/common/soc-acpi-intel-tgl-match.c

```diff
@@ -394,6 +399,20 @@ struct snd_soc_acpi_mach snd_soc_acpi_intel_tgl_machines[] = {
 		.sof_fw_filename = "sof-tgl.ri",
 		.sof_tplg_filename = "sof-tgl-es8336.tplg",
 	},
+	{
+		.id = "10EC1308",
+		.drv_name = "tgl_rt1308_lt6911uxc",
+		.machine_quirk = snd_soc_acpi_codec_list,
+		.quirk_data = &tgl_lt6911_hdmi,
+		.sof_fw_filename = "sof-tgl.ri",
+		.sof_tplg_filename = "sof-tgl-rt1308-HDMI-I2S.tplg"
+	},
+	{
+		.id = "10EC1308",
+		.drv_name = "tgl_rt1308",
+		.sof_fw_filename = "sof-tgl.ri",
+		.sof_tplg_filename = "sof-tgl-rt1308.tplg"
+	},
```

Comment:
I am also not comfortable with this entry, basically for ANY case where this ACPI ID is found, you will enable your own driver. It's a bit risky, I am aware of other internal designs with this amplifier and this would lead to conflicts.

Is this really required, or can you add a more specific matching based on e.g. DMI or some other ACPI device that's present in the platform.

Reply:
This has been added to support the customers who doesn't require HDMI capture in their product. Also TGL is in sustenance phase and not sure how many customer uses only legacy feature(RT1308 amplifier alone). Hence for safe side to kept it.
Also my understanding, On kernel mainline there is no I2S machine check and support for RT1308 in I2S mode and its enabled only in SDW mode. looks no customers other than IOTG uses RT1308 in I2S mode.  As my knowledge if customer doesn't require RT1308 on I2S then will be disabled on BIOS config and ACPI id wont get enumerated anymore. In this case no conflict on driver match.

Please comment if any other use case or scenario which I'm not aware and should be considered.

Reply:
ok. I excluded the generic rt1308 topology loading from my driver since no specific customer product DMI/ACPI. Currently the hard requirement is RT1308+ HDMI I2S config so added only that config based on the Lontium chip codec ID. Hope it will not have any conflict and risk.